src/config.rs

```diff
@@ -14,6 +15,21 @@ const ENV_VAR_CONFIG_PREFIX: &str = "FAMEDLY_LDAP_SYNC";
 /// Separator for setting a list using env vars
 const ENV_VAR_LIST_SEP: &str = " ";
 
+/// Configuration for the sync client
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+pub struct Config {
+	/// Configuration related to Famedly Zitadel
+	pub famedly: FamedlyConfig,
+	/// Optional LDAP configuration
+	pub ldap: Option<LdapConfig>,
+	/// Optional Endpoint configuration
+	pub endpoint: Option<EndpointConfig>,
```

Comment:
This is very minor, but I think it would be nicer to fold `ldap` and `endpoint` into a different top-level config key.

We could also break out the definition for `LdapConfig` and `EndpointConfig` into a different module, that would make the testing here a bit less busy.

Reply:
@tlater-famedly 

I agree to separate the source-specific configs into the source-specific modules. Will do that.

However, I don't agree to fold `ldap` and `endpoint` into a different top-level config key, at least not at this point. It's unnecessary complication and lot's of code rewrite for a benefit that is heavily based on point of view. I think such nesting just obfuscates things.

Reply:
Hm, I don't see the obfuscation or the increased complication. It's just adding another serde struct, which will be a net increase of 2 LOC.

Reply:
Done.